### DISCUSSION TITLE
How does AnyCable handle acknowledgments

### DISCUSSION THREAD

**Discussion by kceb** [anycable/anycable#164] [General]
One of the big drawbacks of ActionCable is lack of ACK for sent messages. Does anycable have delivery guarantees and acknowledgment handling?

I assume yes but I'm wondering how it does it under the hood.

**Comment by palkan** [comment]
> Does anycable have delivery guarantees and acknowledgment handling? I assume yes but I'm wondering how it does it under the hood.

Not yet.
ACKs is a part of the future protocol version (hasn't been published yet).

Delivery guarantees could be achieved by keeping track of received messages (aka "streams history"). The support is coming in v1.4 (experimental) with the plan to release the final version in v1.5. Both version should be released 'till the end of the year.

Some details (and the extended protocol examples) could be seen here: https://noti.st/palkan/MeBUVe/the-pitfalls-of-realtime-ification
